log-manager/src/main/java/io/airlift/log/LogFileName.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.log;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ComparisonChain;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.google.common.base.CharMatcher.anyOf;
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+import static java.util.UUID.randomUUID;
+
+public class LogFileName
+        implements Comparable<LogFileName>
+{
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("-yyyyMMdd.HHmmss");
+
+    private final String fileName;
+    private final LocalDateTime dateTime;
+    private final OptionalInt index;
+    private final OptionalInt legacyIndex;
+    private final Optional<String> slug;
+    private final boolean compressed;
+
+    public static Optional<LogFileName> forName(String baseName, String fileName)
```

Comment:
This needs an explanation -- it's not clear what's the purpose of this method. What's `baseName` vs `fileName`? What does a result of `Optional.empty()` mean? The method is way too complex to expect someone to reverse engineer those answers from it.

Reply:
I added comments and renamed the method. Let me know what you think.